test/unittest/tests/block/test_block_api_write.cpp

```diff
@@ -0,0 +1,279 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * [block_api]: block_write.c
+ * The block manager performs file operations on specific blocks which eventually gets translated
+ * onto files. This test suite aims to test the possible cases for the write(), read()
+ * and write_size() API operations.
+ */
+#include <catch2/catch.hpp>
+#include <iostream>
+#include <filesystem>
+
+#include "wt_internal.h"
+#include "../wrappers/mock_session.h"
+#include "../wrappers/config_parser.h"
+#include "../wrappers/item_wrapper.h"
+
+const int DEFAULT_BLOCK_SIZE = 256;
+const std::string ALLOCATION_SIZE = "256";
+const std::string BLOCK_ALLOCATION = "best";
+const std::string OS_CACHE_MAX = "0";
+const std::string OS_CACHE_DIRTY_MAX = "0";
+const std::string ACCESS_PATTERN = "random";
+const std::string DEFAULT_FILE_NAME = "test.txt";
+
+typedef std::pair<std::array<uint8_t, WT_BTREE_MAX_ADDR_COOKIE>, size_t> addr_cookie;
+
+void
+test_and_validate_write_size(WT_BM *bm, std::shared_ptr<mock_session> session, size_t size)
+{
+    size_t init_size = size;
+    REQUIRE(bm->write_size(bm, session->get_wt_session_impl(), &size) == 0);
+    CHECK(size % DEFAULT_BLOCK_SIZE == 0);
+    CHECK(size == ((init_size / DEFAULT_BLOCK_SIZE) + 1) * DEFAULT_BLOCK_SIZE);
+}
+
+void
+validate_block_contents(WT_BM *bm, std::shared_ptr<mock_session> session, WT_ITEM *write_buf,
+  addr_cookie cookie, wt_off_t offset, uint32_t size)
+{
+    // Test that the write was correctly written via __wt_read().
+    WT_ITEM read_buf;
+    WT_CLEAR(read_buf);
+    REQUIRE(__wt_buf_initsize(session->get_wt_session_impl(), &read_buf, write_buf->memsize) == 0);
+    REQUIRE(
+      __wt_read(session->get_wt_session_impl(), bm->block->fh, offset, size, read_buf.mem) == 0);
+    CHECK(memcmp(write_buf->mem, read_buf.mem, write_buf->size) == 0);
+
+    // Test that the write buffer was correctly via bm->read().
+    REQUIRE(bm->read(bm, session->get_wt_session_impl(), &read_buf, cookie.first.data(),
+              cookie.second) == 0);
+    WT_BLOCK_HEADER *blk = reinterpret_cast<WT_BLOCK_HEADER *>(WT_BLOCK_HEADER_REF(write_buf->mem));
+    blk->checksum = 0;
```

Comment:
a bm->read() clears checksum as a part of testing. I wonder if this is correct behavior, and should be remained cleared.

Reply:
Can you elaborate on this, I think we should be comparing the write checksum with the read checksum but we're not at this stage presumably because of this?

Reply:
As discussed I have created a ticket [WT-13549](https://jira.mongodb.org/browse/WT-13549)